test/framework/dump.go

```diff
@@ -48,188 +47,132 @@ func (f *CommonFramework) DumpDefaultResourcesInAllNamespaces(ctx context.Contex
-	result = multierror.Append(result, err)
-	if err == nil {
-		for _, os := range operatingSystems.Items {
-			f.dumpGardenerExtension(&os)
-		}
-		f.Logger.Println()
-	}
-
-	f.Logger.Infof("%s [EXTENSIONS] [WORKER]", ctxIdentifier)
-	workers := &v1alpha1.WorkerList{}
-	err = k8sClient.Client().List(ctx, workers, client.InNamespace(namespace))
-	result = multierror.Append(result, err)
-	if err == nil {
-		for _, worker := range workers.Items {
-			f.dumpGardenerExtension(&worker)
-		}
-		f.Logger.Println()
-	}
-
-	f.Logger.Infof("%s [EXTENSIONS] [BACKUPBUCKET]", ctxIdentifier)
-	backupBuckets := &v1alpha1.BackupBucketList{}
-	err = k8sClient.Client().List(ctx, backupBuckets, client.InNamespace(namespace))
-	result = multierror.Append(result, err)
-	if err == nil {
-		for _, bucket := range backupBuckets.Items {
-			f.dumpGardenerExtension(&bucket)
-		}
-		f.Logger.Println()
-	}
-
-	f.Logger.Infof("%s [EXTENSIONS] [BACKUPENTRY]", ctxIdentifier)
-	backupEntries := &v1alpha1.BackupEntryList{}
-	err = k8sClient.Client().List(ctx, backupEntries, client.InNamespace(namespace))
-	result = multierror.Append(result, err)
-	if err == nil {
-		for _, entry := range backupEntries.Items {
-			f.dumpGardenerExtension(&entry)
-		}
-		f.Logger.Println()
-	}
-
-	f.Logger.Infof("%s [EXTENSIONS] [BASTION]", ctxIdentifier)
-	bastions := &v1alpha1.BastionList{}
-	err = k8sClient.Client().List(ctx, bastions, client.InNamespace(namespace))
-	result = multierror.Append(result, err)
-	if err == nil {
-		for _, entry := range bastions.Items {
-			f.dumpGardenerExtension(&entry)
-		}
-		f.Logger.Println()
-	}
 
-	f.Logger.Infof("%s [EXTENSIONS] [NETWORK]", ctxIdentifier)
-	networks := &v1alpha1.NetworkList{}
-	err = k8sClient.Client().List(ctx, networks, client.InNamespace(namespace))
-	result = multierror.Append(result, err)
-	if err == nil {
-		for _, network := range networks.Items {
-			f.dumpGardenerExtension(&network)
+	for kind, objList := range map[string]client.ObjectList{
+		"Infrastructure":        &extensionsv1alpha1.InfrastructureList{},
+		"ControlPlane":          &extensionsv1alpha1.ControlPlaneList{},
+		"OperatingSystemConfig": &extensionsv1alpha1.OperatingSystemConfigList{},
+		"Worker":                &extensionsv1alpha1.WorkerList{},
+		"BackupBucket":          &extensionsv1alpha1.BackupBucketList{},
+		"BackupEntry":           &extensionsv1alpha1.BackupEntryList{},
+		"Bastion":               &extensionsv1alpha1.BastionList{},
+		"Network":               &extensionsv1alpha1.NetworkList{},
+	} {
+		log.Info("Dumping extensions.gardener.cloud/v1alpha1 resources", "kind", kind)
+
+		if err := k8sClient.Client().List(ctx, objList, client.InNamespace(namespace)); err != nil {
+			result = multierror.Append(result, err)
+			if err := meta.EachListItem(objList, func(o runtime.Object) error {
+				obj, err := extensions.Accessor(o)
+				if err != nil {
+					return err
+				}
+				f.dumpGardenerExtension(log, obj)
+				return nil
```

Comment:
You could use a logger with `kind` value set for calling `dumpGardenerExtension`
```suggestion
		extensionLog := log.WithValues("kind", kind)
		extensionLog.Info("Dumping extensions.gardener.cloud/v1alpha1 resources")

		if err := k8sClient.Client().List(ctx, objList, client.InNamespace(namespace)); err != nil {
			result = multierror.Append(result, err)
			if err := meta.EachListItem(objList, func(o runtime.Object) error {
				obj, err := extensions.Accessor(o)
				if err != nil {
					return err
				}
				f.dumpGardenerExtension(extensionLog, obj)
				return nil
```